### PR TITLE
Build client reproducible

### PR DESCRIPTION
**Pull request by theSoenke**
Build client in a reproducible way. This includes the generated tarballs. Only the Innosetup installer is not reproducible yet

### CODE REVIEW

File: build/config.sh [hunk 1]
```diff
@@ -7,7 +7,6 @@ pushd $BUILD_DIR > /dev/null
 export GOVERSION=${GOVERSION:-1.10.3}
 export REVISION=${GIT_COMMIT:-$(git rev-parse HEAD)}
 export LIBRARY_REVISION=$(cat Gopkg.lock | grep github.com/phrase/phraseapp-go -A 2 | tail -n 1 | cut -d '"' -f 2)
-export CURRENT_DATE=$(TZ=UTC date +"%Y-%m-%dT%H:%M:%SZ")
```
Comment (phraseapp-vincent) [inline build/config.sh]: introduce `CHANGE_DATE` with last commit as default
`export CHANGE_DATE=${SOURCE_DATE_EPOCH:- $(git show HEAD --format=format:'%ct')}`

File: build/go_build.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ function build {
 	name=$3
 	echo "build os=${goos} arch=${goarch}" > /dev/stderr
 
-	CGO_ENABLED=0 GOOS=$goos GOARCH=$goarch go build -o $bin_dir/${name} -ldflags "-X main.BUILT_AT=$CURRENT_DATE -X=main.REVISION=$REVISION -X=main.PHRASEAPP_CLIENT_VERSION=$VERSION -X=main.LIBRARY_REVISION=$LIBRARY_REVISION -extldflags '-static'" .
+	CGO_ENABLED=0 GOOS=$goos GOARCH=$goarch go build -o $bin_dir/${name} -ldflags "-X=main.REVISION=$REVISION -X=main.PHRASEAPP_CLIENT_VERSION=$VERSION -X=main.LIBRARY_REVISION=$LIBRARY_REVISION -extldflags '-static'" .
```
Comment (phraseapp-vincent) [inline build/go_build.sh]: include the new information here too:
`... -X main.LAST_CHANGE=$CHANGE_DATE`

File: info_command.go [hunk 3]
```diff
@@ -8,7 +8,6 @@ import (
 
 func GetInfo() string {
 	info := []string{
-		fmt.Sprintf("Built at:                            %s", BUILT_AT),
```
Comment (phraseapp-vincent) [inline info_command.go]: `fmt.Sprintf("Last change at:                            %s", LAST_CHANGE),`

File: build/build.sh [hunk 4]
```diff
@@ -18,11 +18,11 @@ rm -f build.tar
 # Homebrew - binary must be called phraseapp, because the binary name inside
 # the tar will be made available system wide
 cp phraseapp_macosx_amd64 phraseapp
-tar cfz phraseapp_macosx_amd64.tar.gz phraseapp
+tar cf --mtime='1970-01-01' phraseapp | gzip -n > phraseapp_macosx_amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (theSoenke) [reply build/build.sh]: gzip is here used explicitly to remove the timestamp with `gzip -n`

File: build/config.sh (hunk 1 above)
Comment (theSoenke) [reply build/config.sh]: Using the date of the last git commit is a good idea

File: build/go_build.sh (hunk 2 above)
Comment (theSoenke) [reply build/go_build.sh]: I think it does not provide that much value as we still include the git commit hash in the build info

Comment (phraseapp-vincent) [reply build/go_build.sh]: It may not provide extra information, but it is more human readable (since its a timestamp)

**Pull request merged**